include/nuttx/mm/mm_map.h

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * include/nuttx/mm/mm_map.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_MM_MM_MAP_H
+#define __INCLUDE_NUTTX_MM_MM_MAP_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/fs/fs.h>
+#include <nuttx/queue.h>
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* Type of the mapping */
+
+enum mm_map_type
+{
+  MM_MAP_ANONYMOUS,
+  MM_MAP_NAMED,
+  MM_MAP_INODE,
+  MM_MAP_SHM
+};
+
+/* Id of the mapping */
+
+union mm_map_id_u
+{
+  FAR const char *name;     /* For MM_MAP_NAMED */
+  FAR struct inode *inode;  /* For MM_MAP_INODE */
+  int shmid;                /* FOR MM_MAP_SHM   */
+  uintptr_t val;
+};
+
+/* A memory mapping list item */
+
+struct mm_map_entry_s
+{
+  FAR struct mm_map_entry *flink;  /* this is used as sq_entry_t */
+  enum mm_map_type type;
+  union mm_map_id_u id;
```

Comment:
User may call close before munmap, so it's better to move munmap callback to mm_map_entry_s and remove type and id field.

Reply:
Ok, that may indeed simplify the generic unmap code in mm/ .

Reply:
Done